core/src/main/java/edu/wpi/grip/core/PipelineEntry.java

```diff
@@ -0,0 +1,61 @@
+package edu.wpi.grip.core;
+
+import edu.wpi.grip.core.sockets.InputSocket;
+import edu.wpi.grip.core.sockets.OutputSocket;
+import edu.wpi.grip.core.sockets.Socket;
+
+import java.util.List;
+import java.util.NoSuchElementException;
+
+/**
+ * An entry in the pipeline.
+ */
+public interface PipelineEntry {
+
+  /**
+   * Gets a read-only list of the input sockets to this entry.
+   * This may be empty, but may never be null.
+   */
+  List<InputSocket> getInputSockets();
+
+  /**
+   * Gets a read-only list of the output sockets from this entry.
+   * This may be empty, but may never be null.
+   */
+  List<OutputSocket> getOutputSockets();
+
+  /**
+   * Gets the socket with the given ID in this entry.
+   *
+   * @param uid the UID of the socket to get
+   *
+   * @throws NoSuchElementException if there is no socket with the given UID,
+   */
+  Socket getSocketByUid(String uid) throws NoSuchElementException;
```

Comment:
~~Perhaps using the UUID type would be nice.~~

Never mind. I understand now.
Is there some sort of launch time guarantee of being unique?